test/base-tests/upload-to-aws.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+set -xeuo pipefail
+
+source /etc/os-release
+
+# Fedora 31's composer-cli doesn't support doing image uploads.
+if [[ $ID == fedora ]] && [[ $VERSION_ID == 31 ]]; then
+    echo "Fedora 31 does not support image uploads with composer-cli."
+    exit 0
+fi
+
+# Colorful output.
+function greenprint {
+    echo -e "\033[1;32m${1}\033[0m"
+}
+
+# Apply lorax patch to work around pytoml issues in RHEL 8.x.
+# See BZ 1843704 or https://github.com/weldr/lorax/pull/1030 for more details.
+if [[ $ID == rhel ]]; then
+    sudo sed -r -i 's#toml.load\(args\[3\]\)#toml.load(open(args[3]))#' \
+        /usr/lib/python3.6/site-packages/composer/cli/compose.py
+    sudo rm -f /usr/lib/python3.6/site-packages/composer/cli/compose.pyc
+fi
+
+# We need jq for parsing composer-cli output.
+if ! hash jq; then
+    greenprint "Installing jq"
+    sudo dnf -qy install jq
+fi
+
+TEST_ID="${CHANGE_ID}-${BUILD_ID}"
+IMAGE_KEY=osbuild-composer-base-test-${TEST_ID}
+
+# Jenkins sets WORKSPACE to the job workspace, but if this script runs
+# outside of Jenkins, we can set up a temporary directory instead.
+if [[ ${WORKSPACE:-empty} == empty ]]; then
+    WORKSPACE=$(mktemp -d)
+fi
+
+# Set up temporary files.
+TEMPDIR=$(mktemp -d)
+AWS_CONFIG=${TEMPDIR}/aws.toml
+BLUEPRINT_FILE=${TEMPDIR}/blueprint.toml
+BLUEPRINT_NAME=targetimage
+COMPOSE_START=${TEMPDIR}/compose-start-${IMAGE_KEY}.json
+COMPOSE_INFO=${TEMPDIR}/compose-info-${IMAGE_KEY}.json
+
+# Get the compose log.
+get_compose_log () {
+    COMPOSE_ID=$1
+    LOG_FILE=${WORKSPACE}/osbuild-${ID}-${VERSION_ID}-aws.log
+
+    # Download the logs.
+    sudo composer-cli compose log $COMPOSE_ID | tee $LOG_FILE > /dev/null
```

Comment:
Again, variable-expansion belongs into quotes if you want it as a single argument.

Reply:
I might be misunderstanding something, but what is the repeated `tee FILE >/dev/null` all about? Why not redirect the original call?

Reply:
I don't know, most of this code is copy pasted because I prefer not to include other bash source files. But I wrote only few lines of this file ¯\_(ツ)_/¯